app/src/androidTest/java/org/mozilla/tv/firefox/FirefoxTestRunner.kt

```diff
@@ -6,11 +6,46 @@ package org.mozilla.tv.firefox
 
 import android.app.Application
 import android.content.Context
+import android.os.Bundle
 import android.support.test.runner.AndroidJUnitRunner
+import org.mozilla.tv.firefox.helpers.FakePocketVideoRepoProvider
+import org.mozilla.tv.firefox.pocket.PocketVideoRepo
+import org.mozilla.tv.firefox.ui.PocketBasicUserFlowTest
+import org.mozilla.tv.firefox.ui.screenshots.PocketErrorTest
+import org.mozilla.tv.firefox.utils.ServiceLocator
 
 class FirefoxTestRunner : AndroidJUnitRunner() {
 
+    private lateinit var app: Application
+
     override fun newApplication(cl: ClassLoader?, className: String?, context: Context?): Application {
-        return super.newApplication(cl, FirefoxTestApplication::class.java.name, context)
+        return super.newApplication(cl, FirefoxTestApplication::class.java.name, context).also {
+            app = it
+        }
+    }
+
+    override fun onCreate(arguments: Bundle?) {
+        when (arguments.extractClass()) {
+            // Set up test class specific ServiceLocator
+            PocketBasicUserFlowTest::class.java.name,
+            PocketErrorTest::class.java.name -> {
```

Comment:
I don't like that we explicitly need to define every test class using subbed deps here. What if we:
```kotlin
class TestRunner {
    val testClass = arguments.extractClass()
    if (testClass is TestDependencyProvider) {
        app.overrideServiceLocator(testClass.testServiceLocator)
    }
}

class WhateverTest : TestDependencyProvider {
    override val testServiceLocator = ...
}
```

That way:
- All the test code is defined (at least initially) within its own file
- We don't have to maintain a giant list of explicit classes separate from the test itself

Reply:
I like that change a lot.  I'll set it up.